otx/algorithms/common/utils/utils.py

```diff
@@ -16,13 +16,22 @@
 
 import importlib
 import inspect
+from collections import defaultdict
 from typing import Callable, Optional, Tuple
 
 import yaml
 
 from otx.api.utils.argument_checks import YamlFilePathCheck, check_input_parameters_type
 
 
+class UncopiableDefaultDict(defaultdict):
+    """Uncopiable defaultdict."""
+
+    def __deepcopy__(self, memo):
+        """deepcopy."""
```

Comment:
I'm not sure but doesn't error occur because first ltter is lower case in pre-commit?

Reply:
No error in pre-commit.